src/Common/src/Interop/Windows/SChannel/UnmanagedCertificateContext.IntPtr.cs

```diff
@@ -0,0 +1,46 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Net.Security;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography.X509Certificates;
+
+namespace System.Net
+{
+    internal static unsafe partial class UnmanagedCertificateContext
+    {
+        internal static X509Certificate2Collection GetRemoteCertificatesFromStoreContext(IntPtr certContext)
+        {
+            X509Certificate2Collection result = new X509Certificate2Collection();
+
+            Interop.Crypt32.CERT_CONTEXT context =
+                Marshal.PtrToStructure<Interop.Crypt32.CERT_CONTEXT>(certContext);
```

Comment:
Rather than the NullReferenceException that occurs when doing PtrToStructure(0), can we add the IntPtr.Zero => empty collection check from the SafeHandle version?

Reply:
Yes, I can do that.